References.php

```diff
@@ -222,4 +222,29 @@ public function interlace(string $interlaceType): string
             'http://www.imagemagick.org/script/command-line-options.php#interlace'
         ));
     }
+
+    /**
+     * Checks that threshold value is valid according to ImageMagick command line reference.
+     */
+    public function threshold(string $threshold): float
+    {
+        $threshold = \trim($threshold);
+
+        if (substr($threshold, -1) == '%') {
+            $percent_num = substr($threshold, 0, -1);
+            if (is_numeric($percent_num))
+                return $threshold;
+        } else {
+            if (is_int(filter_var($threshold, FILTER_VALIDATE_INT)) && intval($threshold) >= 0)
+                return $threshold;
```

Comment:
You can replace your check with a simple `is_numeric()` validation. As stated in the `threshold` option in the [threshold docs](http://www.imagemagick.org/script/command-line-options.php#threshold), the value can be negative or zero, it's an edge-case supported by ImageMagick, and it doesn't throw an error, therefore we shouldn't prevent it.

```suggestion
            if (is_numeric($threshold) && $threshold > 0) {
                return $threshold;
            }
```

Reply:
I've also just realized that this if can actually be attached to the else above it instead of being isolated inside it. It's a minor refactor, I'm just not yet sure how I would commit such change.